File: 프로그래머스/3/49189. 가장 먼 노드/가장 먼 노드.py
from collections import deque

def solution(n, edge):
    edge = sorted(edge)
    d=[0]*(n+1)
    queue=deque([1])
    graph=[[] for _ in range(n+1)]
    answer=0
    
    for i in edge:
        graph[i[0]].append(i[1])
        graph[i[1]].append(i[0])

    d[1]=1
    while queue:
        cur = queue.popleft()
        for node in graph[cur]:
            if d[node] == 0:
                queue.append(node)
                d[node] = d[cur]+1
                
    max_d = max(d)
    for j in d:
        if j == max_d:
            answer+=1
    return answer
                
    

File: 프로그래머스/3/49189. 가장 먼 노드/가장 먼 노드.js
function solution(n, edge) {
    const graph = new Array(n).fill().map(_ => [])
    for(let e of edge){
        graph[e[0]-1].push(e[1]-1)
        graph[e[1]-1].push(e[0]-1)
    }
    const visited = [1]
    const queue = [0]
    while(queue.length){
        const cur = queue.shift()
        for(let next of graph[cur]){
            if(!visited[next]){
                visited[next] = visited[cur]+1
                queue.push(next)
            }
        }
    }
    const max = Math.max(...visited)
    return visited.filter(e=>e === max).length
}